web/public/causality.mjs

```diff
@@ -0,0 +1,48 @@
+import "./d3.v355.min.js";
+import './graphviz.min.js';
+
+export function renderCausality(dot){
+  const foundSvg = d3.select(".causality-graph");
+  const svg = createSvg(foundSvg);
+
+  graphviz.graphviz.layout(dot, "svg", "dot").then(content => {
+    svg.html(content);
+
+    // disable tooltips that graphviz auto-generates
+    svg.selectAll('title').remove();
+    svg.selectAll('*').attr('xlink:title', null);
+
+    var bbox = svg.node().getBBox()
+    d3.select(svg.node().parentNode)
+      .attr("viewBox", "" + (bbox.x - 20) + " " + (bbox.y - 20) + " " + (bbox.width + 40) + " " + (bbox.height + 40))
+  })
+}
+
+function createSvg(svg) {
+  var g = d3.select("g.test")
+  var zoom = d3.behavior.zoom();
+
+  if (g.empty()) {
+    svg.append("defs").append("filter")
+      .attr("id", "embiggen")
+      .append("feMorphology")
+      .attr("operator", "dilate")
+      .attr("radius", "4");
+
+    g = svg.append("g").attr("class", "test")
+    svg.on("mousedown", function() {
+      var ev = d3.event;
+      if (ev.button || ev.ctrlKey)
+        ev.stopImmediatePropagation();
+      }
+    )
+    .call(zoom
+      .scaleExtent([0.5, 10])
+      .on("zoom", function() {
+        var ev = d3.event;
+        g.attr("transform", "translate(" + ev.translate + ") scale(" + ev.scale + ")");
+      })
+    );
```

Comment:
Minor annoyance, but if you zoom/pan on one page, then go to another causality page and try to zoom/pan, it jumps back to the previous state:

https://user-images.githubusercontent.com/26583442/123835347-6a1fc500-d8d6-11eb-8cd0-5bb2e8de7b57.mov

This can be fixed pretty easily here:

```diff
diff --git a/web/public/causality.mjs b/web/public/causality.mjs
index 87b1de232..e980dca77 100644
--- a/web/public/causality.mjs
+++ b/web/public/causality.mjs
@@ -43,6 +43,8 @@ function createSvg(svg) {
         g.attr("transform", "translate(" + ev.translate + ") scale(" + ev.scale + ")");
       })
     );
+    // Reset graph focus in case we previously modified it on another page.
+    zoom.translate([0,0]).scale(1).center(0,0);
   }
   return g;
 }
```

Reply:
This should be fixed with https://github.com/concourse/concourse/commit/d453ff233f45aa8d68fb76bcd583859fb5e0c002#diff-2e1332d21e1a3253c1661d71595bca330a08e4722f22a8f9127be4101537bfccR23 and I don't see it when I do it, is this after hard reloading the page?

Reply:
Ah yeah i guess i had an old version 👍